### PR TITLE
[O2B-1163] Add QC flag verification API

### PR DESCRIPTION
**Pull request by xsalonx**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [x] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- NA

Notable changes for developers:
- Added QC flag verification endpoint

Changes made to the database:
- NA

### CODE REVIEW

File: lib/server/services/qualityControlFlag/QcFlagService.js [hunk 1]
```diff
@@ -300,6 +310,37 @@ class QcFlagService {
...
 
+    /**
+     * Create verification of QC flag
+     * @param {Partial<QcFlagVerification>} parameters flag verification parameters
+     * @param {object} relations QC Flag entity relations
+     * @param {UserWithRoles} relations.userWithRoles user with roles
+     * @return {Promise<QcFlag>} promise
+     * @throws {NotFoundError|AccessDeniedError}
+     */
+    async verifyFlag(parameters, relations) {
```
Comment (martinboulais) [inline lib/server/services/qualityControlFlag/QcFlagService.js:321]: You can't name a parameter parameter... name it verification or whatever you want, but not parameters...

File: lib/server/services/qualityControlFlag/QcFlagService.js [hunk 2]
```diff
@@ -300,6 +310,37 @@ class QcFlagService {
...
+        return dataSource.transaction(async () => {
+            const { flagId, comment } = parameters;
+
+            const qcFlag = await this.getOneOrFail(flagId);
+
+            const { userWithRoles: { userId, externalUserId } } = relations;
```
Comment (martinboulais) [inline lib/server/services/qualityControlFlag/QcFlagService.js:327]: Why expecting roles if you don't use it?

File: test/api/qcFlags.test.js [hunk 3]
```diff
@@ -393,7 +394,7 @@ module.exports = () => {
 
     describe('DELETE /api/qcFlags/:id', () => {
         it('should fail to delete QC flag when being neither owner nor admin', async () => {
```
Comment (martinboulais) [inline test/api/qcFlags.test.js:396]: You may add a test for trying to delete verified flag?

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: lib/server/services/qualityControlFlag/QcFlagService.js (hunk 1 above)
Comment (xsalonx) [reply lib/server/services/qualityControlFlag/QcFlagService.js]: renamed

File: lib/server/services/qualityControlFlag/QcFlagService.js (hunk 2 above)
Comment (xsalonx) [reply lib/server/services/qualityControlFlag/QcFlagService.js]: removed

File: test/api/qcFlags.test.js (hunk 3 above)
Comment (xsalonx) [reply test/api/qcFlags.test.js:409]: added

**Review by martinboulais — APPROVED** [review]

**Pull request merged**